app/services/course/assessment/submission/update_service.rb

```diff
@@ -102,6 +108,23 @@ def edit_submission_path
     edit_course_assessment_submission_path(current_course, @assessment, @submission)
   end
 
+  # Find the questions for this submission without submission_questions.
+  # Build and save! new submission_questions.
+  #
+  # @return[Boolean] If new submission_questions were created.
+  def build_missing_submission_questions
+    questions_with_submission_questions = @submission.submission_questions.map(&:question)
+    questions_without_submission_questions = questions_to_attempt - questions_with_submission_questions
```

Comment:
Think I won't fix this one, splitting it over multiple lines will reduce the readability, and shortening the variable names will also affect comprehension.